src/Lecture2.hs

```diff
@@ -208,8 +264,13 @@ The algorithm of merge sort is the following:
 [1,2,3]
 -}
 mergeSort :: [Int] -> [Int]
-mergeSort = error "TODO"
-
+mergeSort [] = []
+mergeSort [x] = [x]
+mergeSort list = merge (mergeSort left) (mergeSort right)
+  where 
+    halfLen = div (length list) 2
+    left = take halfLen list
+    right = drop halfLen list
```

Comment:
You can use `splitAt` instead of `take + drop` to avoid traversing the first half twice.

Additionally, you can avoid using `length` if you notice that you don't necessarily need to split in the middle. You only need to split it into two lists of approximately the same size.

In that case, you can implement a solution that traverses a list only once and splits its elements into two lists of elements on only even positions and only odd positions.